service/resource/certificate/current.go

```diff
@@ -2,8 +2,35 @@ package certificate
 
 import (
 	"context"
+	"fmt"
+	"path"
+
+	"github.com/giantswarm/microerror"
+	"github.com/spf13/afero"
 )
 
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	r.logger.Log("debug", fmt.Sprintf("reading certificate directory: %s", r.certificateDirectory))
+
+	fileInfos, err := afero.ReadDir(r.fs, r.certificateDirectory)
+	if err != nil {
+		return nil, microerror.Maskf(err, "could not read certificate directory")
+	}
+
+	certificateFiles := []certificateFile{}
+
+	for _, fileInfo := range fileInfos {
+		filePath := path.Join(r.certificateDirectory, fileInfo.Name())
+		fileData, err := afero.ReadFile(r.fs, filePath)
+		if err != nil {
+			return nil, microerror.Maskf(err, "could not read certificate")
+		}
+
+		certificateFiles = append(certificateFiles, certificateFile{
+			path: filePath,
+			data: string(fileData),
```

Comment:
Is there a special reason why we want to have strings here? I think keeping the byte slice would be more native. We can compare and work with byte slices anyway and it has to be written to the FS anyway as a byte slice.

Reply:
No strong feelings either way. I prefer keeping things in strings, for testing.